src/main/java/spoon/reflect/visitor/chain/Predicate.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (C) 2006-2016 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.reflect.visitor.chain;
+
+/**
+ * @param <T> the type of the input to the function
+ */
+public interface Predicate<T> {

```

Comment:
Yes generic.
1) QueryStep chain can be used to process elements which are not inherited from CtElement. I am not sure whether we needs that.
2) Spoon `Filter` has already old meaning like Filtering of children produced by CtScanner, I wanted something more generic, what can be used to decide whether input element should be sent to output or not.